modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -86,6 +86,28 @@
     return new MapStream<>(multimap.entries().stream());
   }
 
+  /**
+   * Returns a stream of map entries where the keys and values are taken from a collection.
+   *
+   * @param <V>  the key and value type
+   * @param collection  the collection
+   * @return a stream of map entries derived from the values in the collection
+   */
+  public static <V> MapStream<V, V> of(Collection<V> collection) {
+    return of(collection.stream());
+  }
+
+  /**
+   * Returns a stream of map entries where the keys and values are taken from a stream.
+   *
+   * @param <V>  the key and value type
+   * @param stream  the stream
+   * @return a stream of map entries derived from the values in the stream
+   */
+  public static <V> MapStream<V, V> of(Stream<V> stream) {
+    return of(stream, key -> key);
```

Comment:
Looking at the current factory methods in MapStream, this should probably be inlined and have the identity function removed.